rdf4j/src/main/java/org/apache/commons/rdf/rdf4j/RDF4JLiteral.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.rdf.rdf4j;
+
+import org.eclipse.rdf4j.model.Literal;
+
+/**
+ * Marker interface for RDF4J implementations of Literal.
+ * 
+ */
+public interface RDF4JLiteral 
+	extends RDF4JTerm<org.eclipse.rdf4j.model.Literal>, org.apache.commons.rdf.api.Literal {
+	
+	/**
+	 * Return the corresponding RDF4J {@link org.eclipse.rdf4j.model.Literal}.
+	 * 
+	 * @return The corresponding RDF4J Literal
+	 */
+	Literal asValue();
```

Comment:
I actaully prefer this method now. If generics are removed from RDF4JTerm it would no longer be a null override and would still return a Literal as expected.